flag_engine/identities/models.py

```diff
@@ -1,18 +1,30 @@
 import datetime
 import typing
 import uuid
+from collections import UserList
 from dataclasses import dataclass, field
 
 from flag_engine.features.models import FeatureStateModel
 from flag_engine.identities.traits.models import TraitModel
+from flag_engine.utils.exceptions import DuplicateFeatureState
+
+
+class IdentityFeaturesList(UserList):
```

Comment:
I'd prefer if this lived somewhere else. Perhaps in a `utils.collections` module or something?

Reply:
yeah, I did thought about that but I wasn't really 100 percent satisfied with any location that I could think of because I how closely tied this is with the model

Reply:
Yeah, I guess it sounds a little bit generic but it would contain classes which subclass classes from python's collections module so it seems to make the most sense? 

We can leave it where it is if you feel strongly about it, I just don't particularly like that it's not a _model_ itself.

Reply:
Hmm yeah it does make sense